src/ui/hooks/recordings.ts

```diff
@@ -174,6 +183,11 @@ export async function getRecordingOwnerUserId(
 export function useGetRecording(
   recordingId: RecordingId | null
 ): { recording: Recording | undefined; isAuthorized: boolean; loading: boolean } {
+  if (isMock()) {
+    const env = mockEnvironment();
+    return env ? env.database.useGetRecording(recordingId) : { loading: true };
```

Comment:
Was this meant to return `{loading: true}` temporarily if the mock environment isn't ready, and then return the actual mock recording afterwards? 

I'm only about 80% sure, but this looks like it will only ever return `{ loading: true }` if it happens to be called while the environment isn't ready. It won't update its returned value once the environment is ready.

Reply:
> It won't update its returned value once the environment is ready.

To be more precise: it doesn't need to update the returned value, but it needs to tell React that the value has changed so that all components using this hook are rerendered. You need something like this:
```typescript
  if (isMock()) {
    const [result, setResult] = useState<any>({
      recording: undefined,
      isAuthorized: false,
      loading: true,
    });
    useEffect(() => {
      async function waitForRecording() {
        const env = await waitForMockEnvironment();
        setResult(env.database.useGetRecording(recordingId));
      }
      waitForRecording();
    });
    return result;
  }
```
(React hooks are the devil in disguise)

Reply:
Yeah, that looks good!

In general, adding hooks (useState/useEffect) inside an if statement is not ideal since React will throw an error if the order that the hooks are called in changes. But in this case, isMock() should always return the same value, and the order will always be the same, so it should be okay. The proper way to do it is a fair bit messier.

Last nit, let's pass in an empty array in the useEffect so that it only gets called once:

```js
    useEffect(() => {
      async function waitForRecording() {
        const env = await waitForMockEnvironment();
        setResult(env.database.useGetRecording(recordingId));
      }
      waitForRecording();
    }, []);
```